cmd/cli/version.go

```diff
@@ -1,32 +1,138 @@
+This command prints out the following version information:
+(1) the OSM version installed in the mesh name and namespace,
+(2) the local OSM CLI version,
+(3) the latest OSM release version available.
+
+The mesh to check for version information is identified
+by its mesh name and namespace.
+Use the --mesh-name and --osm-namespace flags to specify the mesh to
+check for version information.
+`
+
+const versionExample = `
+# Print the osm version in the mesh "osm1" and namespace "osm-ns1".
+# Also print the local CLI version and the latest release version available.
+osm version --mesh-name osm1 --osm-namespace osm-ns1
 `
 
-// PrintCliVersion prints the version
-func PrintCliVersion(out io.Writer) {
-	_, _ = fmt.Fprintf(out, "Version: %s; Commit: %s; Date: %s\n", version.Version, version.GitCommit, version.BuildDate)
+type versionCmd struct {
+	meshName  string
+	remote    bool
+	out       io.Writer
+	config    *rest.Config
+	clientSet kubernetes.Interface
 }
 
+const (
+	osmReleaseRepoOwner = "openservicemesh"
+	osmReleaseRepoName  = "osm"
+)
+
 func newVersionCmd(out io.Writer) *cobra.Command {
+	v := &versionCmd{
+		out: out,
+	}
+
 	cmd := &cobra.Command{
-		Use:   "version",
-		Short: "osm cli version",
-		Long:  versionHelp,
-		Args:  cobra.NoArgs,
-		Run: func(cmd *cobra.Command, args []string) {
-			PrintCliVersion(out)
+		Use:     "version",
+		Short:   "print 1) version installed in mesh, 2) local cli version, 3) latest available version",
+		Long:    versionHelp,
+		Example: versionExample,
+		Args:    cobra.ExactArgs(0),
+		RunE: func(_ *cobra.Command, args []string) error {
+			config, err := settings.RESTClientGetter().ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig: %s", err)
+			}
+			v.config = config
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
+			}
+			v.clientSet = clientset
+			return v.run()
 		},
 	}
+
+	f := cmd.Flags()
+	f.BoolVar(&v.remote, "remote", true, "Use remote=false to skip checking osm version installed in the remote mesh.")
+	f.StringVar(&v.meshName, "mesh-name", defaultMeshName, "Name of the mesh to check for version information")
+
 	return cmd
 }
+
+func (c *versionCmd) run() error {
+	// only check remote mesh version information if remote flag is true
+	if c.remote {
+		meshNamespace := settings.Namespace()
+		osmControllerDeployment, err := getMeshControllerDeployment(c.clientSet, c.meshName, meshNamespace)
+
+		if err != nil {
+			_, _ = fmt.Fprint(c.out, err.Error()+"\n")
+		} else {
+			// get mesh version currently installed
+			meshVersion := osmControllerDeployment.ObjectMeta.Labels[constants.OSMAppVersionLabelKey]
+			if meshVersion == "" {
+				meshVersion = "Unknown"
+			}
+			_, _ = fmt.Fprintf(c.out, "Mesh Name: %s. Mesh Namespace: %s. Mesh Version: %s.\n", c.meshName, meshNamespace, meshVersion)
```

Comment:
nit: The local CLI version information is separated by semi-colons. To be consistent the remote version information could also be separated by semi-colons.